internal/monitors/expvar/expvar.go

```diff
@@ -0,0 +1,393 @@
+package expvar
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/signalfx/golib/datapoint"
+	"github.com/signalfx/signalfx-agent/internal/core/config"
+	"github.com/signalfx/signalfx-agent/internal/monitors"
+	"github.com/signalfx/signalfx-agent/internal/monitors/types"
+	"github.com/signalfx/signalfx-agent/internal/utils"
+	log "github.com/sirupsen/logrus"
+	"io/ioutil"
+	"net/http"
+	url2 "net/url"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+)
+
+const monitorType = "expvar"
+
+const (
+	gauge                                  = "gauge"
+	counter                                = "counter"
+	memstatsPauseNsMetricPath              = "memstats.PauseNs"
+	memstatsPauseEndMetricPath             = "memstats.PauseEnd"
+	memstatsNumGCMetricPath                = "memstats.NumGC"
+	memstatsMostRecentGCPauseNsMetricName  = "memstats.most_recent_gc_pause_ns"
+	memstatsMostRecentGCPauseEndMetricName = "memstats.most_recent_gc_pause_end"
+	memstatsBySizeSizeMetricPath           = "memstats.BySize.Size"
+	memstatsBySizeMallocsMetricPath        = "memstats.BySize.Mallocs"
+	memstatsBySizeFreesMetricPath          = "memstats.BySize.Frees"
+	memstatsBySizeDimensionPath            = "memstats.BySize"
+)
+
+var logger = log.WithFields(log.Fields{"monitorType": monitorType})
+
+// Config for this monitor
+type Config struct {
+	config.MonitorConfig `yaml:",inline" acceptsEndpoints:"true"`
+	// Host of the expvar endpoint
+	Host string `yaml:"host" validate:"required"`
+	// Port of the expvar endpoint
+	Port uint16 `yaml:"port" validate:"required"`
+	// If true, the agent will connect to the host using HTTPS instead of plain HTTP.
+	UseHTTPS bool `yaml:"useHTTPS"`
+	// If useHTTPS is true and this option is also true, the host's TLS cert will not be verified.
+	SkipVerify bool `yaml:"skipVerify"`
+	// Path to the expvar endpoint, usually `/debug/vars` (the default).
+	Path string `yaml:"path" default:"/debug/vars"`
+	// Excludes metrics memstats.alloc, memstats.by_size.size, memstats.by_size.mallocs and memstats.by_size.frees if false
+	EnhancedMetrics bool `yaml:"enhancedMetrics"`
+	// Metrics configurations
+	MetricConfigs []*MetricConfig `yaml:"metrics"`
+	url           *url2.URL
+	runInterval   time.Duration
+}
+
+// MetricConfig for metric
+type MetricConfig struct {
+	// Metric name
+	Name string `yaml:"name"`
+	// JSON path of the metric value
+	JSONPath string `yaml:"JSONPath" validate:"required"`
+	// SignalFx metric type. Possible values are "gauge" or "counter"
+	Type string `yaml:"type" validate:"required"`
+	// Metric dimensions
+	DimensionConfigs []*DimensionConfig `yaml:"dimensions"`
+	metricType       datapoint.MetricType
```

Comment:
Same thing.  Just derive this as needed in the monitor, it isn't that expensive.